File: IronVault/IronVault.Model/Korisnik.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace IronVault.Model
{
    public partial class Korisnik
    {
        public int KorisnikId { get; set; }
        public string Ime { get; set; } = null!;
        public string Prezime { get; set; } = null!;
        public string KorisnickoIme { get; set; } = null!;
        public string? Email { get; set; }
        public string? Slika { get; set; }
        public string? BrojTelefona { get; set; }
        public float Visina { get; set; }
        public float Tezina { get; set; }
        public int GradId { get; set; }
        public int SpolId { get; set; }
        public int TeretanaId { get; set; }

        public virtual ICollection<KorisnikUloga> KorisnikUlogas { get; set; } = new List<KorisnikUloga>();
    }
}

File: IronVault/UI/ironvault_mobile/lib/models/korisnik.dart
import 'package:ironvault_mobile/models/grad.dart';
import 'package:ironvault_mobile/models/spol.dart';
import 'package:json_annotation/json_annotation.dart';

part 'korisnik.g.dart';

@JsonSerializable()
class Korisnik {
  int? korisnikId;
  String? ime;
  String? prezime;
  String? korisnickoIme;
  String? email;
  String? slika;
  String? brojTelefona;
  double? visina;
  double? tezina;
  String? vrijemeUteretani;
  int? razina;
  int? gradId;
  Grad? grad;
  int? spolId;
  Spol? spol;
  String? lozinka;
  String? lozinkaPotvrda;
  int? satiUteretani;

  Korisnik({this.korisnikId});

  // / Connect the generated [_$PersonFromJson] function to the `fromJson`
  // / factory.
  factory Korisnik.fromJson(Map<String, dynamic> json) =>
      _$KorisnikFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$KorisnikToJson(this);
}
